src/builtin.jq

```diff
@@ -304,5 +301,8 @@ def JOIN($idx; stream; idx_expr):
   stream | [., $idx[idx_expr]];
 def JOIN($idx; stream; idx_expr; join_expr):
   stream | [., $idx[idx_expr]] | join_expr;
-def IN(s): reduce (first(select(. == s)) | true) as $v (false; if . or $v then true else false end);
-def IN(src; s): reduce (src|IN(s)) as $v (false; if . or $v then true else false end);
+def IN(s): . as $in | first( if (s == $in) then true else empty end ) // false;
+def IN(src; s): any( src|IN(s); .);
+
+# f should always evaluate to a string:
+def GROUPS_BY(stream; f): reduce stream as $x ({}; .[$x|f] += [$x] ) | .[];
```

Comment:
This should be all lower-case.  Though it's SQLish, it's also jq-ish.

Reply:
@nicowilliams - As I mentioned elsewhere, git and github have me all tied up in knots at the moment, so please make any changes you see fit.  

I used uppercase for GROUPS_BY to emphasize the distinction with the existing `group_by`.  Incidentally, is it too late to re-implement `group_by` so that it does not incur the costs of sorting?  (For example, `group_by` could be implemented using `GROUPS_BY`.)